modules/images/webp-uploads/load.php

```diff
@@ -96,6 +96,31 @@ function webp_uploads_create_sources_property( array $metadata, $attachment_id )
 			continue;
 		}
 
+		$original_image_filesize = isset( $metadata['filesize'] ) ? (int) $metadata['filesize'] : 0;
+		$webp_image_filesize     = isset( $image['filesize'] ) ? (int) $image['filesize'] : 0;
+		if ( $original_image_filesize <= 0 && $webp_image_filesize <= 0 ) {
+			continue;
+		}
+
+		/* Remove WebP image if it bigger then original image */
+		if (
+			$webp_image_filesize >= $original_image_filesize
+			/**
+			 * Filter whether to prioritize the file type over the mime type.
+			 *
+			 * By default the performance lab plugin will use the mime type with the smaller filesize
+			 * rather than defaulting to `webp`.
+			 *
+			 * @since n.e.x.t
+			 *
+			 * @param bool $perfer_filesize Prioritize file size over mime type. Default true.
+			 */
+			&& apply_filters( 'webp_prioritise_filesize_over_mime_type', true )
```

Comment:
Just a suggestion: can we put the filter first and then put the condition to check image file size? or we don't want to run the extra codes from the plugins/themes before we actually check for the file sizes?